WPMedia/fusion-news-theme-blocks
# PEN-1697: Lazy load ads by default

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
Added lazy-loading functionality to Ads Block

## Jira Ticket
- [PEN-1697](https://arcpublishing.atlassian.net/browse/PEN-1697)

## Acceptance Criteria
![image](https://user-images.githubusercontent.com/26662906/106821311-86c71200-6642-11eb-80fc-d7d41e22ea74.png)

## Test Steps
1. In `Fusion-News-Theme` repo, checkout `master` & `git pull`
2. After checking out this feature branch in `fusion-news-theme-blocks`, run `rm -rf ./node_modules && npx lerna clean && npm i && npx lerna bootstrap && cd blocks/header-nav-chain-block && npm i && cd ../..`
3. Run `npx fusion start -f -l @wpmedia/ads-block` in `Fusion-News-Theme`
4. Either use a page that already has a significant amount of content or create a new one. The idea is, we need a certain amount of content to create enough height in the document to test the lazy loading functionality (since it needs to be far enough down the page for you to see it lazy load). I personally just added a Top Table List to the page with a bunch of content (like 20-30 stories). Also open Chrome DevTools to the **Network** tab and in the "filter" text box, put `dfp`. This will allow you to easily see the ad calls being made as each ad unit is lazy-loaded.
5. Now, add one Ads Block of each ad unit type to the page but make sure to put some at the top and some at the bottom of the page. This will allow you to see which ads are loading right away and which are being deferred until scrolling down the page. The ones at the top will load immediately even with lazy loading enabled (since they are in-view on page load). Make sure the `Lazy Load` custom field is enabled.
6. Stage, publish and open in new tab. You should now see that only the ad units that are in-view (or within 200px of being in view) will load initially. The others that are further down the page won't load until you scroll down to them.
7. Now go back to PageBuilder Editor and set all ad units to have `Lazy Load` **disabled**.
8. Stage, publish and open in separate tab. You should now see that all ad units are loaded immediately on page load.

## Effect Of Changes
### Before
Ads did not lazy-load and all of the ad units on the page would be rendered immediately on page load.

### After
Ads now have a custom field option for enabling lazy-loading (enabled by default). When enabled, ad units will only load once the user has scrolled to within `200px` of the element.

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist 
_The reviewer of the PR should copy-paste this template into the review comments on review._
